src/Android/Xamarin.Android/Samples/Augmented Reality/NavigateAR/RoutePlanner.cs

```diff
@@ -0,0 +1,380 @@
+        private const string ServerUrl = "https://www.arcgis.com/sharing/rest";
+        private const string AppClientId = @"lgAdHkYZYlwwfAhC";
+        private const string OAuthRedirectUrl = @"my-ags-app://auth";
+
+        // Permissions and permission request.
+        private readonly string[] _requestedPermissions = { Manifest.Permission.AccessFineLocation };
+        private const int requestCode = 35;
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Navigate in AR";
+
+            CreateLayout();
+
+            RequestPermissions();
+        }
+
+        private void CreateLayout()
+        {
+            // Create and show the layout.
+            SetContentView(ArcGISRuntime.Resource.Layout.NavigateARRoutePlanner);
+
+            // Set up control references.
+            _helpLabel = FindViewById<TextView>(ArcGISRuntime.Resource.Id.helpLabel);
+            _mapView = FindViewById<MapView>(ArcGISRuntime.Resource.Id.mapView);
+            _navigateButton = FindViewById<Button>(ArcGISRuntime.Resource.Id.navigateButton);
+        }
+
+        private async void Initialize()
+        {
+            // Create and show a map.
+            _mapView.Map = new Map(Basemap.CreateImagery());
+
+            try
+            {
+                // Enable location display.
+                _mapView.LocationDisplay.AutoPanMode = LocationDisplayAutoPanMode.Recenter;
+                await _mapView.LocationDisplay.DataSource.StartAsync();
+                _mapView.LocationDisplay.IsEnabled = true;
+
+                // Configure authentication.
+                SetOAuthInfo();
+                var credential = await AuthenticationManager.Current.GenerateCredentialAsync(_routingUri);
+                AuthenticationManager.Current.AddCredential(credential);
+
+                // Create the route task.
+                _routeTask = await RouteTask.CreateAsync(_routingUri);
+
+                // Create route display overlay and symbology.
+                _routeOverlay = new GraphicsOverlay();
+                SimpleLineSymbol routeSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, System.Drawing.Color.Yellow, 1);
+                _routeOverlay.Renderer = new SimpleRenderer(routeSymbol);
+
+                // Create stop display overlay.
+                _stopsOverlay = new GraphicsOverlay();
+
+                // Add the overlays to the map.
+                _mapView.GraphicsOverlays.Add(_routeOverlay);
+                _mapView.GraphicsOverlays.Add(_stopsOverlay);
+
+                // Enable tap-to-place stops.
+                _mapView.GeoViewTapped += MapView_GeoViewTapped;
+
+                // Update the UI.
+                _helpLabel.Text = "Tap to set a start point";
+            }
+            catch (Exception ex)
+            {
+                ShowMessage("Failed to start sample", "Error");
```

Comment:
You can replace the ShowMessage method with
`new AlertDialog.Builder(this).SetMessage("Failed to start sample").SetTitle("Error").Show();`